### PR TITLE
apply clippy linter suggestions

### PR DESCRIPTION
**Pull request by billmakes**
I know this is crazy but I was messing with your project here to learn more Rust, and I've just gotten around to using clippy lints for enhanced linter suggestions. I went through the project and manually applied these suggestions. 

https://rust-lang.github.io/rust-clippy/master/

It was a good refactoring practice on my end feel free to close this if you'd like.

### CODE REVIEW

**Comment by dcts** [comment]
Haha this is amazing! Some changes I really like (using `match` instead of `if`, feels much more rust style). Other changes I kinda dont like (putting too much on a single line). Sometimes I prefere to have a long line broken down into multiple lines for better readibility, but it might be that Im breaking the rust "rules". 

Do you think all improvements from clippy should be always implemented? In the [documentation](https://doc.rust-lang.org/nightly/clippy/) they state:
> A collection of lints to catch common mistakes and improve your Rust code. 

They also have different "lint levels". Did you apply the most strict level maybe? :)

**Comment by dcts** [comment]
Also how did you find my blackjack game? 🤣 I have another cli terminal game in the making (connect-4) 🤣🤣🤣
[github.com/dcts/connect4-cli](https://github.com/dcts/connect4-cli)

File: src/main.rs [hunk 1]
```diff
@@ -30,38 +31,45 @@ pub fn main() {
                     print_game_end(&game, "❌ BUSTED ❌");
                     break;
                 }
-            },
+            }
             PlayerAction::Stop => {
                 // check game state and end conditions
-                if game.dealer_score > game.player_score {
```
Comment (billmakes) [inline src/main.rs:36]: [clippy::comparison_chain](https://rust-lang.github.io/rust-clippy/master/#comparison_chain)

File: src/graphics.rs [hunk 2]
```diff
@@ -113,21 +116,24 @@ pub fn print_cards_hidden(cards: &Vec<Card>) {
         Value::Ten => {
             cards_string = cards_string.replace("v ", cards[0].value_str());
             cards_string = cards_string.replace(" v", cards[0].value_str());
-        },
+        }
         _ => {
-            cards_string = cards_string.replace("v", cards[0].value_str());
```
Comment (billmakes) [inline src/graphics.rs:118]: [clippy::single_char_pattern](https://rust-lang.github.io/rust-clippy/master/#single_char_pattern)

File: src/graphics.rs [hunk 3]
```diff
@@ -93,14 +96,14 @@ pub fn print_cards(cards: &Vec<Card>) {
...
+            final_string.push_str(&split[line - 1]);
         }
-        final_string.push_str("\n");
+        final_string.insert(final_string.len(), '\n');
     }
-    println!("{}",final_string);
+    println!("{}", final_string);
 }
 
-pub fn print_cards_hidden(cards: &Vec<Card>) {
```
Comment (billmakes) [inline src/graphics.rs:103]: [clippy::ptr_arg](https://rust-lang.github.io/rust-clippy/master/#ptr_arg)

File: src/graphics.rs [hunk 4]
```diff
@@ -93,14 +96,14 @@ pub fn print_cards(cards: &Vec<Card>) {
     let mut final_string = String::from("");
     for line in 1..=5 {
         for split in card_splits.iter() {
-            final_string.push_str(&split[line-1]);
+            final_string.push_str(&split[line - 1]);
         }
-        final_string.push_str("\n");
```
Comment (billmakes) [inline src/graphics.rs:98]: [clippy::single_char_add_str](https://rust-lang.github.io/rust-clippy/master/#single_char_add_str)

File: src/card.rs [hunk 5]
```diff
@@ -8,40 +9,13 @@ pub struct Card {
...
-            color: color,
-        }
+        Card { value, color }
     }
 
     pub fn random() -> Card {
         Card::new(random_value(), random_color())
     }
 
-    pub fn to_string(&self) -> String {
```
Comment (billmakes) [inline src/card.rs:21]: [clippy::inherent_to_string](https://rust-lang.github.io/rust-clippy/master/#inherent_to_string)

**Comment by billmakes** [comment]
> Haha this is amazing! Some changes I really like (using `match` instead of `if`, feels much more rust style). Other changes I kinda dont like (putting too much on a single line). Sometimes I prefere to have a long line broken down into multiple lines for better readibility, but it might be that Im breaking the rust "rules".
> 
> Do you think all improvements from clippy should be always implemented? In the [documentation](https://doc.rust-lang.org/nightly/clippy/) they state:
> 
> > A collection of lints to catch common mistakes and improve your Rust code.
> 
> They also have different "lint levels". Did you apply the most strict level maybe? :)

I totally get that! I found your project by just googling rust blackjack cli games haha. I went through the PR with a review and linked the relevant lints. Some changes are surprisingly small or nitpicky but clippy seems to think there is a noteworthy performance boost, or some are simply just a stylistic thing. Do I think they should all be implemented? I honestly have no idea, I was told clippy was a huge help from some experienced rust devs in a discord channel somewhere... You're definitely further along experience wise than me in rust, I'll keep an eye out on your connect4 game! I've just been working through The Rust Programming Language book and lurking around random rust cli apps on github to get a feel for application structure and patterns.

https://github.com/rust-lang/rust-clippy
you can integrate it into vim or vscode or whatever you'd like. I don't have it configured for any linting levels so I believe it's giving me every suggestion, I'm just treating them as a second opinion or a live tip sort of thing while I learn lol - I'll look up the linting rule for more details and explanations on why something is "better" performance wise or stylistically

File: src/game.rs [hunk 6]
```diff
@@ -11,37 +11,33 @@ pub struct Game {
...
-    
+
     pub fn dealer_draw_card(&mut self) {
         self.dealer_cards.push(Card::random());
         self.dealer_score = Game::compute_score(&self.dealer_cards);
     }
-    
-    fn compute_score(cards: &Vec<Card>) -> u8 {
+
+    fn compute_score(cards: &[Card]) -> u8 {
```
Comment (dcts) [inline src/game.rs:40]: @billmakes do you understand this shortcut?

**Comment by dcts** [comment]
@billmakes thanks for sharing clippy, seems super usefull! :)

File: src/game.rs (hunk 6 above)
Comment (billmakes) [reply src/game.rs:40]: Uhh sort of lol. So from my understanding, explicitly requiring a whole reference to a vector collection of type Card restricts the flexibility of your `compute_score` function. Allowing any slice containing Card elements will let your function not only just take in a whole vector collection but any slice of Card elements as well. I'd say this is probably not a big deal at all for the scope of this project but clippy will let you know how it feels either way, it couldn't hurt to keep in mind. There's also something about indirection and how the references are pointing to the memory addresses but it's a bit beyond me haha

I actually recently learned this from the following resources: 

https://rust-lang.github.io/rust-clippy/master/
https://rust-unofficial.github.io/patterns/idioms/coercion-arguments.html#description
https://www.youtube.com/watch?v=hBA00Tx_dIM

**Pull request merged**